src/meta/MetaServiceUtils.cpp

```diff
@@ -516,48 +516,41 @@ std::string MetaServiceUtils::userKey(UserID userId) {
     return key;
 }
 
-std::string MetaServiceUtils::userVal(const std::string& password,
-                                      const cpp2::UserItem& userItem) {
-    auto len = password.size();
-    std::string val, userVal;
-    apache::thrift::CompactSerializer::serialize(userItem, &userVal);
-    val.reserve(sizeof(int32_t) + len + userVal.size());
-    val.append(reinterpret_cast<const char*>(&len), sizeof(int32_t))
-       .append(password)
-       .append(userVal);
+std::string MetaServiceUtils::userVal(const nebula::cpp2::UserItem& userItem) {
+    std::string val;
+    apache::thrift::CompactSerializer::serialize(userItem, &val);
     return val;
 }
 
-folly::StringPiece MetaServiceUtils::userItemVal(folly::StringPiece rawVal) {
-    auto offset = sizeof(int32_t) + *reinterpret_cast<const int32_t *>(rawVal.begin());
-    return rawVal.subpiece(offset, rawVal.size() - offset);
-}
+std::string MetaServiceUtils::replaceUserVal(const nebula::cpp2::UserItem& user,
```

Comment:
how about change the parameter name like user=>newUser, val=>oldUserVal

Reply:
> how about change the parameter name like user=>newUser, val=>oldUserVal

Aha, here I using thrift object direct. So I can modifying individual user property value when thrift val was set. And serialize it for storage. Thrift object cab be backward compatible.